xdsl/ir.py

```diff
@@ -265,6 +266,13 @@ def verify(self) -> None:
         """
         pass
 
+    def __str__(self) -> str:
```

Comment:
THis way, we have the danger of having those lines all over the project right?
```bash
        from xdsl.printer import Printer
        res = StringIO()
        printer = Printer(stream=res)
```

Reply:
I am not sure how a different solution would look like. We will need to instantiate the Printer at some point. Maybe the import could be moved to the top? This could make circular dependencies though...

Reply:
I guess the other way is moving the lines starting from [here](https://github.com/xdslproject/xdsl/blob/main/xdsl/printer.py#L284) out of the printer and into the individual attributes. Actually in terms of software engineering, this might even be a cleaner solution. Differentiating among MLIR and xDSL syntax can still be done in the Printer. One just uses this `str` method, the other still uses the old printing infrastructure...

Reply:
I don't really understand your comment George. Do you mean that we risk duplicating this code?
From what I understand, the point of having it here is to only write `print(my_attribute)`.
Though this is maybe not what you meant, so please correct me!

Reply:
Maybe there is indeed a better way of doing that by putting the printing/parsing code inside attributes.
However, we need to be careful, since we can parse or print an attribute both in a generic format and a custom format.
But otherwise, maybe that would be a better option? (probably for another PR though)

Reply:
Yes, I worry about whether this will lead to duplications. Not sure I have a better solution in my mind, though.

Reply:
Well, I would argue that this removes code duplication, since I don't see another case where we would use these lines, since now we can just write `str(my_attr)` instead of this code.

Reply:
If this is good for you for now, I would suggest that we merge it, and we can go back at it if we realize that we duplicate too much code!